mass_mailing_switzerland/models/mass_mailing_contact.py

```diff
@@ -79,8 +122,33 @@ def action_update_to_mailchimp(self):
         """
         if self.env.context.get("skip_mailchimp"):
             return True
-        return super(MassMailingContact,
-                     self.with_context(lang="en_US")).action_update_to_mailchimp()
+
+        out = True
+
+        for contact_to_update in self:
+            # if previous write failed reference to mailchimp member will be lost. error 404
+            try:
+                out = out and super(MassMailingContact,
+                                    contact_to_update.with_context(lang="en_US")).action_update_to_mailchimp()
+            except Exception as e:
+                # if not contact were fond. means an error occur at last write().
+                # Email field in odoo and mailchimp are now different.
+                # solution : we remove previous link to mailchimp and export the contact with new mail
+                if e.args[0] and literal_eval(e.args[0])['status'] == 404:
+                    available_mailchimp_lists = self.env['mailchimp.lists'].search([])
+                    lists = available_mailchimp_lists.mapped('odoo_list_id').ids
+
+                    contact_to_update.subscription_list_ids.filtered(
+                        lambda x: x.list_id.id in lists).write({"mailchimp_id": False})
+                # raise exception if it's any other type
+                else:
+                    self.env.clear()
```

Comment:
The `clear` is not at the right place. No need to clear if we simply raise the error.
```suggestion
```